olmo/model.py

```diff
@@ -473,20 +474,42 @@ def reset_parameters(self):
             self.k_norm.reset_parameters()
         if self.q_norm is not None:
             self.q_norm.reset_parameters()
-        init_weights(
-            self.config,
-            self.attn_out,
-            d=self.config.d_model,
-            layer_id=self.layer_id,
-            type_of_module=ModuleType.out_module,
-        )
-        init_weights(
-            self.config,
-            self.ff_out,
-            d=self.ff_out.in_features,
-            layer_id=self.layer_id,
-            type_of_module=ModuleType.out_module,
-        )
+
+        if self.config.init_fn == InitFnType.normal:
+            init_normal(self.attn_out, std=self.config.init_std, init_cutoff_factor=self.config.init_cutoff_factor)
+            init_normal(self.ff_out, std=self.config.init_std, init_cutoff_factor=self.config.init_cutoff_factor)
+
+            # TODO: what is up with this extra divisor? We may not want this.
+            # TODO: This potentially came from trying to include full_megatron init.
+            # git blame rabbit hole:
+            #   https://github.com/allenai/OLMo/pull/239/commits/0c5b7b5ccac26c3e4729322520f1fdd5b0a4d54e#diff-14cdafe7342088f0c4851cc670185044902088ba6bc5ddcf0a464d698e4165fe
+            #   https://github.com/allenai/OLMo/commit/ba20a857fde0c2fb2e8dbcba07e3781fe088ba06
+            # This is for the case of ff_out._is_residual = True
+            with torch.no_grad():
+                self.ff_out.weight.div_(math.sqrt(2 * self.config.n_layers))
```

Comment:
This was copied from the original MPT code, and now I'm thinking we should remove this unless there's precedent somewhere else, like megatron.